modules/aggregation-layers/src/contour-layer/contour-layer.js

```diff
@@ -43,48 +45,56 @@ const defaultProps = {
   zOffset: 0.005
 };
 
-export default class ContourLayer extends CompositeLayer {
+// props , when changed requires re aggregation (weightmap generation)
+const AGGREGATION_PROPS = ['gpuAggregation'];
+
+export default class ContourLayer extends GridAggregationLayer {
   initializeState() {
-    const {gl} = this.context;
-    const options = {
-      id: `${this.id}-gpu-aggregator`
-    };
-    this.state = {
+    super.initializeState(AGGREGATION_PROPS);
+    this.setState({
       contourData: {},
-      gridAggregator: new GPUGridAggregator(gl, options),
       colorTrigger: 0,
       strokeWidthTrigger: 0
-    };
+    });
+    const attributeManager = this.getAttributeManager();
+    attributeManager.add({
+      positions: {size: 3, accessor: 'getPosition', type: GL.DOUBLE, fp64: false},
+      count: {size: 3, accessor: 'getWeight'}
+    });
   }
 
-  updateState({oldProps, props, changeFlags}) {
-    let dataChanged = false;
+  updateState(opts) {
+    super.updateState(opts);
     let contoursChanged = false;
-    const aggregationFlags = this._getAggregationFlags({oldProps, props, changeFlags});
-    if (aggregationFlags) {
-      dataChanged = true;
-      // Clear countsData cache
+    const dataChanged = this._isAggregationDirty(opts);
+    const cellSizeChanged = opts.oldProps.cellSize !== opts.props.cellSize;
```

Comment:
Aggregator needs `dataChanged` and `cellSIzeChanged` flags seperately, so in CPU aggregation path, projection is not performed when only cell size is changed, `cellSize` can be added to `aggregationProps`, once CPU Aggregation is moved out of aggregator.